pysma/__init__.py

```diff
@@ -122,9 +129,21 @@ def __init__(self, add_default_sensors=True):
                     #   Old 6100_0046C200 is the power generated by the inverter (always equal to grid_power)
                     Sensor("6380_40251E00_0", "pv_power_a", "W"),
```

Comment:
This is the correct pv_power according to #31

Reply:
The thing is that the previous pv_power was using the 6100_0046C200 key and gave me the following entity.

```json
            {
                "entity_id": "sensor.pv_power",
                "config_entry_id": null,
                "device_id": null,
                "area_id": null,
                "unique_id": "sma-6100_0046C200-pv_power",
                "platform": "sma",
                "name": null,
                "icon": null,
                "disabled_by": null,
                "capabilities": {},
                "supported_features": 0,
                "device_class": null,
                "unit_of_measurement": "W",
                "original_name": "pv_power",
                "original_icon": null
            },
```

When the pv_power was removed the import of old yaml started failing because the _check_sensor_schema is checking if all keys in the config exist.

https://github.com/rklomp/home-assistant-core/blob/6985a25a3d243be750fd4ac757c49a10c131d3e9/homeassistant/components/sma/sensor.py#L45-L48

This would break the import of yaml for all users that have pv_power in their config. I used the aliases to fix that
https://github.com/rklomp/home-assistant-core/blob/sma_config_flow/homeassistant/components/sma/sensor.py#L43-L49


In the new import code I am translating the old unique ID into the new unique ID format. For that to work I need the original ID of pv_power, so I can lookup the correct entity.
https://github.com/home-assistant/core/blob/3cd08b045fc83fc8417b087448dc8d1ec9b7b31e/homeassistant/components/sma/config_flow.py#L168-L182

This will give me my new pv_power sensor:
```json
            {
                "entity_id": "sensor.pv_power",
                "config_entry_id": "2283b6456dfea16d41b2ba8f77c70b60",
                "device_id": "4a6e110fee8a60344d1ece487cc178ae",
                "area_id": null,
                "unique_id": "3006894981-6100_0046C200_0",
                "platform": "sma",
                "name": null,
                "icon": null,
                "disabled_by": null,
                "capabilities": {},
                "supported_features": 0,
                "device_class": null,
                "unit_of_measurement": "W",
                "original_name": "grid_power_alternative",
                "original_icon": null
            },
```

I am not sure if using these aliases is the way to go. Any alternative ideas are welcome :)

Another possibility is to just create a translation table in the `/homeassistant/components/sma/const.py` file. And use that to lookup the sensor pv_power in the yaml config with key 6100_0046C200  and translate it to new key 6380_40251E00.
That way no aliases are needed in pysma, as it something that I do not see a future use for.

Just not sure what the best way is to handle this. Let me know your thoughts

Reply:
If it really is just pv_power causing issues, we can rollback the fix for #31, or just rename the sensor to `inverter_power` if that would help?

The translation table would be very clean though, we could put it in pysma as a const as well. This is also something that can easily be removed later

Reply:
If we revert #31 we should also keep the same name as the lookup is done by name.

I'll have a look if I can come up with a clean solution using a translation table for legacy sensors.

Reply:
I have replaced it by a single dictionary I am using in the import.

See the related commit in home assistant:
https://github.com/home-assistant/core/pull/48003/commits/f44b6c0efaee1377b3d0866fd3e84308a3883e69